modules/persistent-storage-csi-migration-vsphere.adoc

```diff
@@ -0,0 +1,83 @@
+// Module included in the following assemblies:
+//
+// * storage/container_storage_interface/persistent-storage-csi-migration.adoc
+
+:_content-type: CONCEPT
+[id="persistent-storage-csi-migration-sc-vsphere_{context}"]
+= vSphere automatic migration
+
+=== New installations of {product-title}
+For new installations of {product-title} 4.13, or later, automatic migration is enabled by default.
+
+=== Upgrading {product-title}
+When upgrading from {product-title} 4.12, or earlier, to 4.13, automatic CSI migration for vSphere only occurs if you opt in.
+
+Note the following consequences when opting in to migration:
+
+* Enabling migration cannot be undone.
+
+* Migration can take awhile to complete depending on how many nodes are on the cluster.
+
+* Migration is a disruptive process. The Machine Config Operator (MCO) has to update kubelet, which means degrading each node as it rolls out the new MachineConfig to each machine.
```

Comment:
Does this migration have to be done; during an upgrade? Or can you upgrade to 4.13 (and then turn on the migration at a later date)?

Reply:
Yes that is possible and that is what we are advocating here actually. It will change in 4.14 though, where migration will happen automatically in 4.14

Reply:
I indicate that you have to opt-in. You do not have to migrate.